docs/programming-oak.md

```diff
@@ -175,118 +176,92 @@ as a gRPC server:
   of the Node, available because the Node `struct` implements the gRPC service
   trait.
 
-## Using an Oak Application from a client
+## Starting an Oak Application
 
-A client that is outside of the Oak ecosystem can use an Oak Application by
-interacting with it as a gRPC service. However, this does involve a few
-additional complications beyond the normal gRPC boilerplate, which are described
-in this section.
+In order run an Oak infrastructure ISV should compile an Oak application to a
+set of WebAssembly modules, serialize them into a binary application
+configuration file and pass it to the Oak Server.
+ISV should also publish a gRPC service endpoint (host:port) for a newly loaded
+Oak Application.
 
-### Connecting to the Oak Manager
+### Creating a Configuration File
 
-Oak client code (in C++) first needs to connect to the
-[Oak Manager](concepts.md#oak-manager), as a normal gRPC client:
+In order to load an Oak Application into the Oak Server its configuration should
+be serialized into a binary file.
+ISV first need to specify a configuration file:
 
 <!-- prettier-ignore-start -->
-[embedmd]:# (../examples/abitest/client/abitest.cc C++ /.*Connect to the Oak Manager/ /NewStub.*;/)
-```C++
-  // Connect to the Oak Manager.
-  auto channel =
-      grpc::CreateChannel(absl::GetFlag(FLAGS_manager_address), grpc::InsecureChannelCredentials());
-  auto manager_stub = oak::Manager::NewStub(channel, grpc::StubOptions());
+[embedmd]:# (../examples/hello_world/config/config.textproto /node_configs.*/ /initial_entrypoint_name.*/)
 ```
-<!-- prettier-ignore-end -->
-
-This sets up a client for the `oak.Manager` service; this service allows the
-creation and termination of Oak Applications:
-
-<!-- prettier-ignore-start -->
-[embedmd]:# (../oak/proto/manager.proto protobuf /^service Manager/ /^}/)
-```protobuf
-service Manager {
-  // Request the creation of a new Oak Application with the specified configuration.
-  //
-  // After the Oak Node is created, the client should connect to the returned
-  // endpoint via gRPC and perform a direct attestation against the Node itself,
-  // to verify that its configuration corresponds to what the client expects.
-  rpc CreateApplication(CreateApplicationRequest) returns (CreateApplicationResponse);
-
-  // Request that an Oak Application terminate.
-  rpc TerminateApplication(TerminateApplicationRequest) returns (TerminateApplicationResponse);
+node_configs {
+  name: "app"
+  wasm_config {
+    module_bytes: "<bytes>"
+  }
 }
+node_configs {
+  name: "log"
+  log_config {}
+}
+grpc_port: 8080
+initial_node_config_name: "app"
+initial_entrypoint_name: "oak_main"
 ```
 <!-- prettier-ignore-end -->
 
-Alternatively, the manager client can also be wrapped up in a
-[helper library](../oak/client/manager_client.h), which will make application
-configuration (described in the next section) easier:
+The `module_bytes: "<bytes>"` means that this value will be filled with
```

Comment:
"will be filled" &ndash; by whom?  Is that something that the `app_config_serializer` does?